packages/base/src/3dview/mainview.tsx

```diff
@@ -461,27 +462,48 @@ export class MainView extends React.Component<IProps, IStates> {
     this.resizeCanvasToDisplaySize();
   };
 
-  private lookAtPosition(position: { x: number; y: number; z: number }) {
-    const objPosition = new THREE.Vector3(
+  private lookAtPosition(
+    position: { x: number; y: number; z: number } | [number, number, number]
+  ) {
+    this._targetPosition = new THREE.Vector3(
       position[0],
       position[1],
       position[2]
     );
-    if (this._camera) {
-      this._camera.lookAt(objPosition);
-      const cameraToTargetDistance =
-        this._camera.position.distanceTo(objPosition);
-
-      if (cameraToTargetDistance < 1) {
-        // Move the camera back slightly to ensure visibility
-        this._camera.position.add(
-          this._camera.getWorldDirection(new THREE.Vector3()).multiplyScalar(-2)
+
+    if (this._camera && this._targetPosition) {
+      const direction = new THREE.Vector3()
+        .subVectors(this._targetPosition, this._camera.position)
+        .normalize();
+
+      // Handle zoom for perspective camera
+      if (this._camera instanceof THREE.PerspectiveCamera) {
+        this._camera.position.copy(
+          this._targetPosition.clone().sub(direction.multiplyScalar(10))
         );
       }
+
+      // Handle zoom for orthographic camera
+      if (this._camera instanceof THREE.OrthographicCamera) {
+        this._camera.zoom = 100;
```

Comment:
This looks off. Should you remove this line? 

e.g. it will not work well with big objects (reference length > 100)

Reply:
We could maybe rely on the bounding box size of the selected object for this. But let's track this in an issue and not block this PR for this 👍🏽

Reply:
Done, i'll look into it.